### PR TITLE
Close #18063 (permuted in numpy frontend)

### LINKED ISSUE #18063: permuted

### PR DESCRIPTION
**Pull request by PerfectionistAF**
Close #18063

### CODE REVIEW

**Comment by PerfectionistAF** [comment]
# permuted test 
### (path: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions)

@handle_frontend_test(
    fn_tree="ivy.functional.frontends.numpy.random.functions.permuted",
    dtype_value=helpers.dtype_and_values(
        available_dtypes=helpers.get_dtypes("valid", full=True),
        shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape")
    ),
)
def test_permuted(
    *,
    dtype_value,
    permutation,
    x,
    out,
    copy,
    test_flags,
    fn_name,
):
    dtype, value = dtype_value
    shape = dtype_value.shape
    dims = [x for x in range(len(shape))]
    permutation = shape(st.permutations(dims))

    helpers.test_frontend_function(
        input_dtypes=dtype,
        test_flags=test_flags,
        fn_name=fn_name,
        x=value[0],
        axes=permutation,
        out=dtype,
        copy=None
    )

**Comment by PerfectionistAF** [comment]
> # permuted test
> ### commit from closed test PR
> ### (path: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions)
> ```
> @handle_frontend_test(
>     fn_tree="ivy.functional.frontends.numpy.random.functions.permuted",
>     dtype_value=helpers.dtype_and_values(
>         available_dtypes=helpers.get_dtypes("valid", full=True),
>         shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape")
>     ),
> )
> ```
> 
> ```
> def test_permuted(
>     *,
>     dtype_value,
>     permutation,
>     x,
>     out,
>     copy,
>     test_flags,
>     fn_name,
> ):
> ```
> 
> ```
>     dtype, value = dtype_value
>     shape = dtype_value.shape
>     dims = [x for x in range(len(shape))]
>     permutation = shape(st.permutations(dims))
>     helpers.test_frontend_function(
>         input_dtypes=dtype,
>         test_flags=test_flags,
>         fn_name=fn_name,
>         x=value[0],
>         axes=permutation,
>         out=dtype,
>         copy=None
>     )
> ```

@PerfectionistAF
Thank you for your contributions, may I ask if you could link the relevant issue this PR is associated with? That way I can provide more structured help.

Also, please could you also comment "add_frontend_checklist" (no quotation marks). A bot should automatically edit your message to include the relevant checklist that you can also edit! 😄

**Comment by PerfectionistAF** [comment]
add_frontend_checklist

File: ivy/functional/frontends/numpy/random/functions.py [hunk 1]
```diff
@@ -175,3 +175,13 @@ def rayleigh(scale, size=None):
     log_u = ivy.log(u)
     x = ivy.multiply(scale, ivy.sqrt(ivy.multiply(-2, log_u)))
     return x
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def permuted(x, axes=None, out=None):#, copy=None):
+    if out != x:  #when out is not x or is None, shuffle in the result
```
Comment (zhumakhan) [inline ivy/functional/frontends/numpy/random/functions.py:182]: `out`'s should not be checked against anything. If out is not None, just need to store result in out and return it

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Comment by zhumakhan** [comment]
Hi, could you please add tests for it?

**Comment by PerfectionistAF** [comment]
The test is pasted above as a comment since some issues occurred with me while committing twice to the same PR

> # permuted test
> ### commit from closed test PR
> ### (path: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions)
> ```
> @handle_frontend_test(
>     fn_tree="ivy.functional.frontends.numpy.random.functions.permuted",
>     dtype_value=helpers.dtype_and_values(
>         available_dtypes=helpers.get_dtypes("valid", full=True),
>         shape=st.shared(helpers.get_shape(min_num_dims=1), key="value_shape")
>     ),
> )
> ```
> 
> ```
> def test_permuted(
>     *,
>     dtype_value,
>     permutation,
>     x,
>     out,
>     copy,
>     test_flags,
>     fn_name,
> ):
> ```
> 
> ```
>     dtype, value = dtype_value
>     shape = dtype_value.shape
>     dims = [x for x in range(len(shape))]
>     permutation = shape(st.permutations(dims))
>     helpers.test_frontend_function(
>         input_dtypes=dtype,
>         test_flags=test_flags,
>         fn_name=fn_name,
>         x=value[0],
>         axes=permutation,
>         out=dtype,
>         copy=None
>     )
> ```

File: ivy/functional/frontends/numpy/random/functions.py [hunk 2]
```diff
@@ -177,6 +177,14 @@ def rayleigh(scale, size=None):
     return x
 
 
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def permuted(x, axes=None, out=None):#, copy=None):
```
Comment (zhumakhan) [inline ivy/functional/frontends/numpy/random/functions.py:182]: It seems this function is not implemented properly. Could you please fix it?

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**